stable-25-2-1: ydb/tests/compatibility/test_ctas.py.TestCTASOperations.test_ctas_olap[rolling_prestable-25-2_to_prestable-25-3]

(pre)stable-25-2 + (pre)stable-25-3


https://github.com/ydb-platform/ydb/actions/runs/17304795448

https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/17304795448/ya-main-x86-64/try_1/ya-test.html#FAIL

ydb/tests/compatibility/test_ctas.py.TestCTASOperations.test_ctas_olap[rolling_prestable-25-2_to_prestable-25-3] 

```
ydb/tests/compatibility/test_ctas.py:45: in test_ctas_olap
    for _ in self.roll():
ydb/tests/library/compatibility/fixtures.py:268: in roll
    node.stop()
ydb/tests/library/harness/kikimr_runner.py:262: in stop
    super(KiKiMRNode, self).stop()
ydb/tests/library/harness/daemon.py:194: in stop
    if not self.__check_can_launch_stop("stop"):
ydb/tests/library/harness/daemon.py:171: in __check_can_launch_stop
    raise DaemonError(
E   ydb.tests.library.harness.daemon.DaemonError: Daemon failed with message: Unexpectedly finished before stop.
E   Process exit_code = -6.
E   Stdout file name: 
E   /home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/compatibility/test-results/py3test/testing_out_stuff/test_ctas.py.TestCTASOperations.test_ctas_olap.rolling_prestable-25-2_to_prestable-25-3/cluster/node_7/stdout
E   Stderr file name: 
E   /home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/compatibility/test-results/py3test/testing_out_stuff/test_ctas.py.TestCTASOperations.test_ctas_olap.rolling_prestable-25-2_to_prestable-25-3/cluster/node_7/stderr
E   Stderr content:
E   
E   GRpc memory quota was set but disabled due to issues with grpc quoter, to enable it use EnableGRpcMemoryQuota option
E   VERIFY failed (2025-08-28T20:37:32.154354Z): DetachActor(7543738337285679618) called for an unknown actor
E   ydb/library/actors/core/mailbox_lockfree.cpp:184
E   DetachActor(): requirement false failed
E   0. /-S/util/system/yassert.cpp:83: InternalPanicImpl @ 0x97B56BA
E   1. /-S/util/system/yassert.cpp:55: Panic @ 0x97B0AEB
E   2. /tmp//-S/ydb/library/actors/core/mailbox_lockfree.cpp:184: DetachActor @ 0xA2885F5
E   3. /tmp//-S/ydb/library/actors/core/executor_thread.cpp:90: UnregisterActor @ 0xA268577
E   4. /tmp//-S/ydb/core/kqp/session_actor/kqp_temp_tables_manager.cpp:239: Finish @ 0x1857E176
E   5. /tmp//-S/ydb/core/kqp/session_actor/kqp_temp_tables_manager.cpp:195: RemoveNextDir @ 0x1857E176
E   6. /tmp//-S/ydb/core/kqp/session_actor/kqp_temp_tables_manager.cpp:234: HandleRemoveDir @ 0x1857EC49
E   7. /tmp//-S/ydb/core/kqp/session_actor/kqp_temp_tables_manager.cpp:92: RmDirsState @ 0x1857EC49
E   8. /tmp//-S/ydb/library/actors/core/executor_thread.cpp:269: Execute @ 0xA2692E4
E   9. /tmp//-S/ydb/library/actors/core/executor_thread.cpp:460: operator() @ 0xA26CD42
E   10. /tmp//-S/ydb/library/actors/core/executor_thread.cpp:512: ProcessExecutorPool @ 0xA26C910
E   11. /tmp//-S/ydb/library/actors/core/executor_thread.cpp:538: ThreadProc @ 0xA26D532
E   12. /-S/util/system/thread.cpp:244: ThreadProxy @ 0x97B9787
E   13. ??:0: ?? @ 0x7FEC4AE26AC2
E   14. ??:0: ?? @ 0x7FEC4AEB884F

logsdir:
/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/compatibility/test-results/py3test/testing_out_stuff
log:
/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/compatibility/test-results/py3test/testing_out_stuff/test_ctas.py.TestCTASOperations.test_ctas_olap.rolling_prestable-25-2_to_prestable-25-3.log
```

<!-- nemesis:related-versions:start -->

---

### Related issues

- Also filed (stable-26-3-1): #49095

<!-- nemesis:related-versions:end -->

--- a/ydb/core/kqp/session_actor/kqp_temp_tables_manager.cpp
+++ b/ydb/core/kqp/session_actor/kqp_temp_tables_manager.cpp
@@ -119,6 +119,7 @@ private:
         const NSchemeCache::TSchemeCacheNavigate* navigate = ev->Get()->Request.Get();
         if (navigate->ErrorCount != 0) {
             Finish();
+            return;
         }
 
         for (const auto& entry : navigate->ResultSet) {
@@ -229,6 +230,7 @@ private:
     void HandleRemoveDir(TEvPrivate::TEvRemoveDirResult::TPtr& result) {
         if (!result->Get()->Result.Success()) {
             Finish();
+            return;
         }
 
         RemoveNextDir();

